src/components/web3/SendCryptoPopup.tsx

```diff
@@ -125,6 +128,18 @@ const highlightBoxCSS = css`
   padding: 16px;
 `;
 
+const onlyNumberOrDecimal = (e: React.KeyboardEvent<HTMLInputElement>) => {
+  const charCode = e.key.charCodeAt(0);
+  if (
+    charCode > 31 &&
+    (charCode < 48 || charCode > 57) &&
+    charCode !== 46 &&
+    e.key != "Backspace"
```

Comment:
I don't think we should forbid backspace in case the user types something incorrectly (I've not actually tested it so not certain what happens).
Does this still allow pasting from the clipboard?
Perhaps we just don't show NaN when the textbox is erased?

Reply:
OK I think I misread the backspace part, but still I'm not sure this is what we should be doing.

Reply:
The code essentially only allows digits + `.` to be typed in the input box. You are right that this prevents pasting, which I will look into. 651df37ff44c8c4e31cd9653c98e6e79bd4412d7 has additional checking on the value, in particular, the `value` is now a string and the popup will refuse to send an outgoing request if `value` is not a proper (parsable) string.